sycl/include/CL/sycl/id.hpp

```diff
@@ -76,31 +63,45 @@ template <int dimensions = 1> class id : public detail::array<dimensions> {
+  id<dimensions> operator op(const id<dimensions> &rhs) const {                \
+    id<dimensions> result;                                                     \
+    for (int i = 0; i < dimensions; ++i) {                                     \
+      result.common_array[i] = this->common_array[i] op rhs.common_array[i];   \
     }                                                                          \
-    id<dimensions> operator op(const size_t &rhs) const {                      \
-      id<dimensions> result;                                                   \
-      for (int i = 0; i < dimensions; ++i) {                                   \
-        result.common_array[i] = this->common_array[i] op rhs;                 \
-      }                                                                        \
-      return result;                                                           \
+    return result;                                                             \
+  }                                                                            \
+  id<dimensions> operator op(const size_t &rhs) const {                        \
+    id<dimensions> result;                                                     \
+    for (int i = 0; i < dimensions; ++i) {                                     \
+      result.common_array[i] = this->common_array[i] op rhs;                   \
     }                                                                          \
-    friend id<dimensions> operator op(const size_t &lhs,                       \
-                                      const id<dimensions> &rhs) {             \
-      id<dimensions> result;                                                   \
-      for (int i = 0; i < dimensions; ++i) {                                   \
-        result.common_array[i] = lhs op rhs.common_array[i];                   \
-      }                                                                        \
-      return result;                                                           \
+    return result;                                                             \
+  }                                                                            \
+  friend id<dimensions> operator op(const size_t &lhs,                         \
+                                    const id<dimensions> &rhs) {               \
+    id<dimensions> result;                                                     \
+    for (int i = 0; i < dimensions; ++i) {                                     \
+      result.common_array[i] = lhs op rhs.common_array[i];                     \
     }                                                                          \
-
+    return result;                                                             \
+  }                                                                            \
+  id<dimensions> operator op(const range<dimensions> &rhs) const {             \
```

Comment:
Are these new operators with range parameters in scope of "implicit conversion from id<1> to size_t" change?

Reply:
Without adding them explicitly, the added conversion makes them ambiguous. See https://gitlab.devtools.intel.com/SYCL/features/issues/20#note_2207506 .

Reply:
In the case of using the implicit conversion, the ambiguous behavior of the math operators is possible: OP(id, range) and OP(size_t, range). 
Specialization of math operators is needed to avoid the ambiguous operator usage